src-ui.v3/src/PkgId.hs

```diff
@@ -157,6 +172,39 @@ data Matches = Matches
   deriving (Eq,Ord)
 
 matchesEmpty :: Matches
-matchesEmpty = Matches { matchesInput = T.empty, matchesExact = Map.empty, matchesInfix = Map.empty} 
+matchesEmpty = Matches { matchesInput = T.empty, matchesExact = Map.empty, matchesInfix = Map.empty}
+
+---------------------------------
+
+data FragRoute = RouteHome
```

Comment:
I'd probably put FragRoute and decodeFrag in Routing.hs now. 

Or at least put encode/decode in the same file so that both are updated at once if they change.